hollow/src/main/java/com/netflix/hollow/api/producer/HollowIncrementalProducer.java

```diff
@@ -36,40 +37,73 @@
 import java.util.concurrent.ConcurrentHashMap;
 
 /**
- * 
+ *
  * Warning: This is a BETA API and is subject to breaking changes.
- * 
+ *
  */
 public class HollowIncrementalProducer {
-    
+
     private static final Object DELETE_RECORD = new Object();
 
     private final HollowProducer producer;
     private final ConcurrentHashMap<RecordPrimaryKey, Object> mutations;
-    
+    private final double threadsPerCpu;
+
     public HollowIncrementalProducer(HollowProducer producer) {
+        this(producer, 1.0d);
+    }
+
+    public HollowIncrementalProducer(HollowProducer producer, double threadsPerCpu) {
         this.producer = producer;
+        this.threadsPerCpu = threadsPerCpu;
         this.mutations = new ConcurrentHashMap<RecordPrimaryKey, Object>();
     }
-    
+
     public void restore(long versionDesired, BlobRetriever blobRetriever) {
         producer.hardRestore(versionDesired, blobRetriever);
     }
-    
+
     public void addOrModify(Object obj) {
         RecordPrimaryKey pk = producer.getObjectMapper().extractPrimaryKey(obj);
         mutations.put(pk, obj);
     }
-    
+
     public void delete(Object obj) {
         RecordPrimaryKey pk = producer.getObjectMapper().extractPrimaryKey(obj);
         delete(pk);
     }
-    
+
     public void delete(RecordPrimaryKey key) {
         mutations.put(key, DELETE_RECORD);
     }
-    
+
+    public static HollowIncrementalProducer.Builder withProducer(HollowProducer producer) {
+        HollowIncrementalProducer.Builder builder = new HollowIncrementalProducer.Builder();
+        return builder.withProducer(producer);
+    }
+
+    public static class Builder {
```

Comment:
Do we need a builder just for one additional argument? Also, constructor for creating incremental producer is public. Just in my opinion does not justify builder pattern. Consider removing it.